### PR TITLE
Render tees for online friends in friends list, change friend info text to `Map | Mode | Country/Ping`, show official server icon, show confirmation popup

### LINKED ISSUE #6614: Render tees for online friends in friends list
Now that online friends in the friends list correspond exactly to players currently on servers (#6612), we can use the tee skin server info to render the tees also in the friends list.

### LINKED ISSUE #6622: Confirmation when joining friends while in race
Would be cool if you need to confirm that you want to join your friend by double clicking in the friends list when you are racing.
The same logic as if you want to disconnect while racing, after 20? minutes you need to confirm it.

### PR DESCRIPTION
**Pull request by Robyt3**
Closes #6614.

friends

Show race disconnect confirmation when joining friend. Closes #6622.

Add popup to confirm connecting to friend on non-official server:

popup

## Checklist

- [X] Tested the change ingame
- [X] Provided screenshots if it is a visual change
- [ ] Tested in combination with possibly related configuration options
- [ ] Written a unit test (especially base/) or added coverage to integration test
- [ ] Considered possible null pointers and out of bounds array indexing
- [ ] Changed no physics that affect existing maps
- [ ] Tested the change with [ASan+UBSan or valgrind's memcheck](https://github.com/ddnet/ddnet/#using-addresssanitizer--undefinedbehavioursanitizer-or-valgrinds-memcheck) (optional)

### CODE REVIEW

**Comment by def-** [comment]
I guess we should take this in for 17.0?

**Comment by Robyt3** [comment]
Yeah. I'll still want to add an icon for official servers. It was also suggested to add a confirmation popup when joining a non-official server with double-click.

**Comment by def-** [comment]
Ok, I'll still push out 17.0-rc1 later if I can get the build working, then we can add this for rc2 when it's ready.

File: src/game/client/components/menus_browser.cpp [hunk 1]
```diff
@@ -585,9 +585,14 @@ void CMenus::RenderServerbrowserServerList(CUIRect View)
 	}
 }
 
-void CMenus::Connect(const char *pAddress)
+void CMenus::Connect(const char *pAddress, std::optional<bool> Official)
 {
-	if(Client()->State() == IClient::STATE_ONLINE && Client()->GetCurrentRaceTime() / 60 >= g_Config.m_ClConfirmDisconnectTime && g_Config.m_ClConfirmDisconnectTime >= 0)
+	if(Official.has_value() && !Official.value())
```
Comment (def-) [inline src/game/client/components/menus_browser.cpp:590]: Hm, i thought the symbol for official server would be enough. Do we really need to suppress unofficial servers like this?

Comment (def-) [reply src/game/client/components/menus_browser.cpp:590]: Screenshot_2023-05-19_at_17 37 44
Blue symbol = "official" (DDNet or KoG tab actually)
I think that's enough and that's why I asked for the blue symbol, so we don't have yet another popup.

**Comment by heinrich5991** [comment]
Wait, why do we add more stuff to "verified" servers? I thought we wanted to get rid of that. Can we get rid of the popup? This makes it seem like we moderate name faking seriously…

**Comment by def-** [comment]
> Wait, why do we add more stuff to "verified" servers?

Because it is useful.

**Pull request merged**